pkg/controller/cloud/node_controller.go

```diff
@@ -36,11 +36,26 @@ import (
 	clientretry "k8s.io/client-go/util/retry"
 	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/klog"
-	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
-	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 	nodeutil "k8s.io/kubernetes/pkg/util/node"
 )
 
+const (
+	// annotationProvidedIPAddr is a node IP annotation set by the "external" cloud provider.
+	// When kubelet is started with the "external" cloud provider, then
+	// it sets this annotation on the node to denote an ip address set from the
+	// cmd line flag (--node-ip). This ip is verified with the cloudprovider as valid by
+	// the cloud-controller-manager
+	// originated from https://github.com/kubernetes/kubernetes/blob/35cf6b6cbc/pkg/kubelet/apis/well_known_annotations.go#L25
+	annotationProvidedIPAddr = "alpha.kubernetes.io/provided-node-ip"
+
+	// taintExternalCloudProvider sets this taint on a node to mark it as unusable,
+	// when kubelet is started with the "external" cloud provider, until a controller
+	// from the cloud-controller-manager intitializes this node, and then removes
+	// the taint
+	// originated from https://github.com/kubernetes/kubernetes/blob/35cf6b6cbc/pkg/scheduler/api/well_known_labels.go#L64
+	taintExternalCloudProvider = "node.cloudprovider.kubernetes.io/uninitialized"
```

Comment:
Maybe move this to `k8s.io/cloud-provider` instead and replace remaining references to point to that too?